### PR TITLE
EVG-6190 modify patch route

### PR DESCRIPTION
**Pull request by ablack12**
Originally our projects patch route merely updated the document in the DB, whereas an update from the UI verifies input, creates a webhook, and enables PR testing and the commit queue.

This version of the PR has no tests, because I'm not sure the best way to test the webhooks piece of this code (which is the main consideration). Would we want to put it in staging and hit the route to verify? Or is it enough if the existing UI route proves unbroken since the logic is the same?

### CODE REVIEW

**Comment by john-m-liu** [comment]
One of your code changes doesn't compile :(

File: rest/data/interface.go [hunk 1]
```diff
@@ -90,6 +90,12 @@ type Connector interface {
 	CreateProject(projectRef *model.ProjectRef) error
 	UpdateProject(projectRef *model.ProjectRef) error
 
+	// EnableWebhooks creates a webhook for the project's owner/repo if one does not exist.
+	// If unable to setup the new webhook, returns false but no error.
+	EnableWebhooks(*model.ProjectRef) (bool, error)
```
Comment (tychoish) [inline rest/data/interface.go]: can this just take projectIDs? do we need the refs?

File: rest/data/interface.go [hunk 2]
```diff
@@ -90,6 +90,12 @@ type Connector interface {
...
+	// EnablePRTesting determines if PR testing can be enabled for the given project.
+	EnablePRTesting(projectRef *model.ProjectRef) error
```
Comment (tychoish) [inline rest/data/interface.go]: I wouldn't name the variable here.

File: rest/data/project.go [hunk 3]
```diff
@@ -55,6 +59,79 @@ func (pc *DBProjectConnector) UpdateProject(projectRef *model.ProjectRef) error
...
+	}
+	return nil
+}
+
+func (pc *DBProjectConnector) EnableCommitQueue(projectRef *model.ProjectRef, commitQueueParams model.CommitQueueParams) error {
+	resultRef, err := model.FindOneProjectRefWithCommitQueueByOwnerRepoAndBranch(projectRef.Owner, projectRef.Repo, projectRef.Branch)
+	if err != nil && !adb.ResultsNotFound(err) {
+		return errors.Wrapf(err, "")
+	}
+	if resultRef != nil && resultRef.CommitQueue.Enabled && resultRef.Identifier != projectRef.Identifier {
```
Comment (tychoish) [inline rest/data/project.go]: can this be a method on resultRef

File: model/github_hook.go [hunk 4]
```diff
@@ -57,3 +64,51 @@ func FindGithubHook(owner, repo string) (*GithubHook, error) {
 
 	return hook, nil
 }
+
+func SetupNewGithubHook(settings evergreen.Settings, owner string, repo string) (*GithubHook, error) {
```
Comment (tychoish) [inline model/github_hook.go]: make caller's pass a context into this to avoid using background.

File: rest/data/interface.go (hunk 1 above)
Comment (ablack12) [reply rest/data/interface.go]: We need repo, owner, and projectID, so I think passing the ref is cleaner than passing 3 strings separately.

File: rest/data/project.go (hunk 3 above)
Comment (ablack12) [reply rest/data/project.go]: Sure

**Pull request merged**